internal/curves/linear.go

```diff
@@ -16,25 +17,52 @@ var (
 
 type LinearSpeedCurve struct {
 	Config configuration.CurveConfig `json:"config"`
-	Value  int                       `json:"value"`
+	Value  float64                   `json:"value"`
+}
+
+func (c *LinearSpeedCurve) Init() {
+	cfg := c.Config.Linear
+	if len(cfg.Steps) > 0 {
+		cfg.FloatSteps = make(map[int]float64)
+
+		for temp, origstr := range cfg.Steps {
+			str := strings.TrimSpace(origstr)
+			l := len(str)
+			isPercent := false
+			if l > 1 && str[l-1] == '%' {
+				isPercent = true
+				str = str[:l-1] // cut off '%' because ParseFloat() wouldn't like it
+			}
+			speed, err := strconv.ParseFloat(str, 64)
+			if err != nil {
+				ui.Warning("Invalid curve step value '%s' in %s", origstr, c.Config.ID)
+			} else {
+				if isPercent {
+					// convert from 1..100 percent to 0..255
+					speed = speed * 2.55
```

Comment:
Ah I see, its still always working with `[0..255]` internally.
But in that case, was the change to `float` even necessary?

Reply:
yes, because 50% is 127.5, for example